### PR TITLE
Porting EdgeNetwork layer to Torch

### PR DESCRIPTION
**Pull request by riya-singh28**
## Description

This PR is to port EdgeNetwork Layer from Tensorflow to PyTorch. Two utilities are added in the pytorch_utils.py namely, unsorted_segment_sum and segment_sum. This PR is rebased version of https://github.com/deepchem/deepchem/pull/3258 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by ARY2260 — APPROVED** [review]
LGTM, @rbharath please provide the review check.

**Review by ARY2260 — CHANGES_REQUESTED** [review]
make the docs locally to verify if the changes are reflected in the docs.

**Comment by riya-singh28** [comment]
EdgeNetwork Layer ->
image

Pytorch utilities ->

image

**Comment by riya-singh28** [comment]
@ARY2260 @rbharath Ready for review.

**Review by ARY2260 — APPROVED** [review]
LGTM, @rbharath, could you please have a look at this PR.

File: deepchem/utils/pytorch_utils.py [hunk 1]
```diff
@@ -14,3 +14,65 @@ def get_activation(fn: Union[Callable, str]):
...
+
+    Parameters
+    ----------
+    data: torch.Tensor
+        A tensor whose segments are to be summed.
+    segment_ids: torch.Tensor
+        The segment indices tensor.
+    num_segments: int
+        The number of segments.
+
```
Comment (rbharath) [inline deepchem/utils/pytorch_utils.py:30]: Can you add a usage example here?

File: deepchem/utils/pytorch_utils.py [hunk 2]
```diff
@@ -14,3 +14,65 @@ def get_activation(fn: Union[Callable, str]):
...
+def segment_sum(data, segment_ids):
+    """Analogous to tf.segment_sum (https://www.tensorflow.org/api_docs/python/tf/math/segment_sum).
+
+    Parameters
+    ----------
+    data: torch.Tensor
+        A pytorch tensor of the data for segmented summation.
+    segment_ids: torch.Tensor
+        A 1-D tensor containing the indices for the segmentation.
+
```
Comment (rbharath) [inline deepchem/utils/pytorch_utils.py:60]: Can you add a usage example here?

File: deepchem/utils/pytorch_utils.py [hunk 3]
```diff
@@ -14,3 +14,65 @@ def get_activation(fn: Union[Callable, str]):
     if isinstance(fn, str):
         return getattr(torch.nn.functional, fn)
     return fn
+
+
+def unsorted_segment_sum(data, segment_ids, num_segments):
```
[the remark this answers is not in the document]
Comment (rbharath) [reply deepchem/utils/pytorch_utils.py:19]: To utils.rst

**Review by rbharath — COMMENTED** [review]
Let's split this PR up into 3 smaller PRs:

1. Unsorted segment sum + tests + docs
2. Segment sum + tests + docs
3. Edge network layer + tests + docs

**Pull request closed without merging**